### ISSUE TITLE
Support for rust programming language

### ISSUE THREAD

**Issue by WaDelma** [testmycode/tmc-langs#12]
Support for rust could be added.
https://www.rust-lang.org/

**Comment by jamox** [comment]
I actually like this proposal. I feel like this could make to a really nice project - do you feel like starting to work on adding support for this language?

**Comment by WaDelma** [comment]
Yeah I would like to start working on it. Not sure how fast it's going progress due other courses. Also this should be probaply split in multiple tasks.

**Comment by jamox** [comment]
Awesome!
I see a few things we need to figure out on the rust side, and a bunch of tasks for the tmc-langs.
I have added some of the tasks here.
Let's plan this a bit more before creating a bunch of issues and a milestone for this.

### Rust:
- [ ] Figure out a way to nicely implement the available points, test method relation; like the `@Points` annotation in java.
- [ ] Maybe a niceish lib for making testing nicer e.g. read/write stdin - maybe reflection like "hey, do you have method X, ok cool you do. let's call it with these params"

> In Java it takes in a string of different point names, where each point is separated by white space. e.g `@Points("124.1 124.2 124.3 124.4 test-point")`
And in C we have `tmc_register_test(s, test_sum, "1 test-point")`

I see at least following sub tasks:
- [ ] New maven module (I wouldn't include it to the tmc-langs-util just yet, but we can include it to the main pom.xml..) ((other products depend on the tmc-langs-util to include all langs modules))
- [ ] Compiling code from langs
- [ ] Running tests from langs
- [ ] Parsing test results from langs
- [ ] Generating the STUB, MODEL SOLUTIONS etc. (we have another issue refactoring this atm.)
- [ ] Having figuring out available points

**Comment by jamox** [comment]
@nygrenh @Mession you were creating the C and python support for langs. Did I overlook some tasks; do you have anything to add?

**Comment by ljleppan** [comment]
Aside from implementing all the methods specified in [`LanguagePlugin`](https://github.com/testmycode/tmc-langs/blob/master/tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/LanguagePlugin.java) (most likely by extending [`AbstractLanguagePlugin`](https://github.com/testmycode/tmc-langs/blob/master/tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/AbstractLanguagePlugin.java)), the plugin also has to implement a [`StudentFilePolicy`](https://github.com/testmycode/tmc-langs/blob/master/tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/io/StudentFilePolicy.java).
